pkg/controllermanager/controller/credentialsbinding/reconciler.go

```diff
@@ -131,6 +153,28 @@ func (r *Reconciler) delete(ctx context.Context, credentialsBinding *securityv1a
 		return nil
 	}
 
+	kind := credentialsBinding.CredentialsRef.Kind
+	credential, errCredentials := kubernetesutils.GetCredentialsByObjectReference(ctx, r.Client, credentialsBinding.CredentialsRef)
+	if errCredentials != nil && !apierrors.IsNotFound(errCredentials) {
+		return errCredentials
+	} else if errCredentials == nil {
```

Comment:
```suggestion
	} else {
```

Nit: Else should be sufficient in this case?!

Reply:
What if it is a not found error?